### ISSUE TITLE
[RFC] Modify the sierra_items_to_dynamo app to use a VHS instead of DynamoDB

### ISSUE THREAD

**Issue by alexwlchan** [wellcometrust/platform#2489]
With strongly typed Sierra numbers, the items_to_dynamo app can't read anything off its queue. Why? Because we serialise the item records using `DynamoFormat[SierraItemRecord]` when they're saved to DynamoDB, but we have no way to deserialise them when getting them from a queue. We could try to find a way to unpack a JSON string using `DynamoFormat`, but it's going to be a pain to test.

I think we should modify the items_to_dynamo app to use a VHS.

Then serialisation is all in JSON, we can ditch a bunch of custom DynamoDB logic, and we get neater DynamoDB tables.

**Comment by alicefuzier** [comment]
I think it makes sense. 
Maybe the quick fix is to write a circe decoder copying the code from DynamoFormat but it's a bit of a crappy solution. I also don't think it would be massively difficult to make items_to_dynamo migrate to VHS at this point and a full re-harvest is a good opportunity to do that.

**Comment by alexwlchan** [comment]
> the quick fix is to write a circe decoder copying the code from DynamoFormat but it's a bit of a crappy solution

We want less custom serialisation code, not more! 😜 

> I also don't think it would be massively difficult to make items_to_dynamo migrate to VHS at this point and a full re-harvest is a good opportunity to do that.

My thinking exactly.

**Comment by kenoir** [comment]
I tend to agree that it's probably a reasonable thing to do, and not that big a change. All the other options I can think of involve hacks to get it to work. 

I defer to @alicefuzier about whether to go ahead though.
